libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -229,15 +231,14 @@ public async Task<InvokeResponse> ProcessActivityAsync(ClaimsIdentity identity,
                 // the Bot will return a specific body and return code.
                 if (activity.Type == ActivityTypes.Invoke)
                 {
-                    Activity invokeResponse = context.TurnState.Get<Activity>(InvokeReponseKey);
-                    if (invokeResponse == null)
+                    var activityInvokeResponse = context.TurnState.Get<Activity>(InvokeReponseKey);
+                    if (activityInvokeResponse == null)
                     {
-                        // ToDo: Trace Here
-                        throw new InvalidOperationException("Bot failed to return a valid 'invokeResponse' activity.");
+                        return new InvokeResponse { Status = (int)HttpStatusCode.NotImplemented };
                     }
                     else
                     {
-                        return (InvokeResponse)invokeResponse.Value;
+                        return (InvokeResponse)activityInvokeResponse.Value;
```

Comment:
Why the cast here? Shouldn't the cast on line 234 be changes to: 
`var activityInvokeResponse = context.TurnState.Get<InvokeResponse>(InvokeReponseKey);`

Reply:
No its actually an Activity on the TurnState. The InvokeResponse is hanging off the Activity. There is no change here - this was intentional.

I was hoping to clarify this confusing contract with the variable name change.

Reply:
The change here is around returning 501 from the HTTP request. That's the point of this change.